lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -376,34 +1017,64 @@ deriveHashable_enum_hashValue(TypeChecker &tc, Decl *parentDecl,
   //     return index.hashValue
   //   }
   // }
+  //
+  // enum SomeEnumWithAssociatedValues {
+  //   case A, B(Int), C(String, Int)
+  //   @derived var hashValue: Int {
+  //     var result: Int
+  //     switch self {
+  //     case A:
+  //       result = _mixInt(0)
```

Comment:
That makes sense! There are lots of to-dos in that Hashing.swift file, so I know it's on the "when there's time for it" pile for the standard library. cc @airspeedswift 

Part of the `Set` implementation is that it has an unusual requirement—two sets can be equal even if their elements are in different orders, and equal sets must have the same hash value. Using XOR to merge mixed hash values solves that, since XOR is commutative/associative in the way that a multiplicative hash wouldn't be (I think).